packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -194,9 +232,56 @@ class CircularProgressIndicator extends ProgressIndicator {
         painter: new _CircularProgressIndicatorPainter(
           valueColor: _getValueColor(context),
           value: value,
-          animationValue: animationValue
+          headValue: headValue,
+          tailValue: tailValue,
+          stepValue: stepValue,
+          rotationValue: rotationValue
         )
       )
     );
   }
 }
+
+class _CircularProgressIndicatorState extends State<CircularProgressIndicator> {
+  Animation<double> _animation;
+  AnimationController _controller;
+
+  RepeatingCurveTween _strokeHeadTween =
+      new RepeatingCurveTween(
+          curve:new Interval(0.0, 0.5, curve: Curves.fastOutSlowIn),
+          repeats:5);
+
+  RepeatingCurveTween _strokeTailTween =
+      new RepeatingCurveTween(
+          curve:new Interval(0.5, 1.0, curve: Curves.fastOutSlowIn),
+          repeats:5);
```

Comment:
I'd probably move these out of the class and make them `final`  They can be shared by all instances of this class.